docs/docs/reference/extend/extension-methods.md

```diff
@@ -0,0 +1,247 @@
+---
+layout: doc-page
+title: "Extension Methods"
+---
+
+Extension methods allow one to add methods to a type after the type is defined. Example:
+
+```scala
+case class Circle(x: Double, y: Double, radius: Double)
+
+implicit object CircleOps {
+  def circumference(this c: Circle): Double = c.radius * math.Pi * 2
+}
+```
+
+`CircleOps` adds an extension method `circumference` to values of class `Circle`. Like regular methods, extension methods can be invoked with infix `.`:
+
+```scala
+  val circle = Circle(0, 0, 1)
+  circle.circumference
+```
+
+Extension methods are methods that have a `this` modifier for the first parameter.
+They can also be invoked as plain methods. So the following holds:
+```scala
+assert(circle.circumference == CircleOps.circumference(circle))
+```
+
+
+
+### Translation of Calls to Extension Methods
+
+When is an extension method considered? There are two possibilities. The first (and recommended one) is by defining the extension method as a member of an implicit value. The method can then be used as an extension method wherever the implicit value is applicable. The second possibility is by making the extension method itself visible under a simple name, typically by importing it. As an example, consider an extension method `longestStrings` on `String`. We can either define it like this:
+
+
+```scala
+implicit object StringSeqOps1 {
+  def longestStrings(this xs: Seq[String]) = {
+    val maxLength = xs.map(_.length).max
+    xs.filter(_.length == maxLength)
+  }
+}
+```
+Then
+```scala
+List("here", "is", "a", "list").longestStrings
+```
+is legal everywhere `StringSeqOps1` is available as an implicit value. Alternatively, we can define `longestStrings`
+as a member of a normal object. But then the method has to be brought into scope to be usable as an extension method.
+
+```scala
+implicit object StringSeqOps2{
```

Comment:
`StringSeqOps2` was meant to be a normal object, so I assume, `implicit` modifier shouldn't be there